src/warc2zim/content_rewriting/html.py

```diff
@@ -95,10 +52,25 @@ def send(self, value: str):
         self.output.write(value)  # pyright: ignore[reportOptionalMemberAccess]
 
     def handle_starttag(self, tag: str, attrs: AttrsList, *, auto_close: bool = False):
-        self._active_tag = tag
         if tag == "script":
-            if "json" in (extract_attr(attrs, "type") or ""):
-                self._active_tag = "json"
+            script_type = self.extract_attr(attrs, "type")
+            if script_type == "json":
+                self.html_rewrite_context = "json"
+            elif script_type == "module":
+                self.html_rewrite_context = "js-module"
+            else:
+                self.html_rewrite_context = "js-classic"
+        elif tag == "link":
+            self.html_rewrite_context = "link"
```

Comment:
This is tricky.
You have removed the previous early `self._active_tag = tag` (the logic was by default _active_tag is tag except special case) and move to the logic "handle special case else falldown to `self.html_rewrite_context = tag` **except** here when you have a default first then special case.

It would be better to always have the same logic. Either default first and we override with special case; or if/elif/else logic, but we need a `else` all the time.

Reply:
Agreed